go113_test.go

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+// +build go1.13
+
+package multierr_test
+
+import (
+	"errors"
+	"os"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/multierr"
+)
+
+type errGreatSadness struct{ id int }
+
+func (errGreatSadness) Error() string {
+	return "great sadness"
+}
+
+type errUnprecedentedFailure struct{ id int }
+
+func (errUnprecedentedFailure) Error() string {
+	return "unprecedented failure"
+}
+
+func (e errUnprecedentedFailure) Unwrap() error {
+	return errRootCause{e.id}
+}
+
+type errRootCause struct{ id int }
+
+func (errRootCause) Error() string {
+	return "root cause"
+}
+
+func TestErrorsWrapping(t *testing.T) {
+	err := multierr.Append(
```

Comment:
As the docstring for `As` contains 
> As attempts to find the first error in the error list

Should we add a test that validates in a case of a multierr As with  multiple possible matches, we return the first one?

Reply:
good call. adding